### ISSUE TITLE
server_test.rb failing `test_malformed_queries` subtest

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#3375]
The `./test-runner server_test.rb` test is failing the `test_malformed_queries ` subtest on `next` with:

```ruby
test_malformed_queries(ClientTest) [/Users/larkost/Projects/rethinkdb/test/rql_test/server_test.rb:133]:
<{"t"=>16, "b"=>[], "r"=>["Protocol error, connection closed."]}> expected but was
<{"t"=>16, "r"=>["Client is buggy (failed to deserialize query)."], "b"=>[]}>.
```

I talked with @Tryneus and he thinks this might be due to a problem with a recent change to the ruby driver. Over to him for that investigation.

**Comment by Tryneus** [comment]
This was caused by a change to the server where it no longer considers a token of '-1' to be a fatal error.  The error here was in the test not being updated, fix is up in review 2512.

**Comment by Tryneus** [comment]
The fix has been approved and merged to `next` in commit 812417b and `v1.16.x` in commit 1320912.
